src/main/java/teammates/ui/webapi/action/ArchiveCourseAction.java

```diff
@@ -25,8 +26,10 @@ public void checkSpecificAccessControl() {
 
     @Override
     public ActionResult execute() {
-        String idOfCourseToArchive = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
-        String archiveStatus = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ARCHIVE_STATUS);
+        CourseArchiveRequest courseArchiveRequest = getAndValidateRequestBody(CourseArchiveRequest.class);
+
+        String idOfCourseToArchive = courseArchiveRequest.getCourseId();
```

Comment:
The courseId is used to identify the resource. So it should not be inside request body.